cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -349,6 +360,7 @@ void update(DatasetId instance, Map<String, String> properties) throws Exception
    *  have {@link Action#ADMIN} privileges on the #instance
    */
   void drop(DatasetId instance) throws Exception {
```

Comment:
During `dropAll` - let's make the enforce check on all the datasets first, and then start dropping them